python-frontend/src/main/java/org/sonar/python/semantic/ClassSymbolImpl.java

```diff
@@ -73,22 +74,23 @@ public ClassSymbolImpl(ClassDef classDef, @Nullable String fullyQualifiedName, P
   }
 
   public ClassSymbolImpl(String name, @Nullable String fullyQualifiedName) {
-    this(name, fullyQualifiedName, null, false, false, null);
+    this(name, fullyQualifiedName, null, false, false, null, false);
   }
 
   public ClassSymbolImpl(String name, @Nullable String fullyQualifiedName, @Nullable LocationInFile definitionLocation,
-                         boolean hasDecorators, boolean hasMetaClass, @Nullable String metaclassFQN) {
+                         boolean hasDecorators, boolean hasMetaClass, @Nullable String metaclassFQN, boolean supportsGenerics) {
```

Comment:
An alternative solution would have been to compute this value lazily. Instead of having a constructor argument, we would compute `supportsGenerics` only when `supportsGenerics()` is called by looking at the super classes.

Reply:
This might be a bit problematic, because when constructing TypeShed symbols we don't add `Generic` superclass in order to avoid having incomplete hierarchies. 
https://github.com/SonarSource/sonar-python/blob/be0952ea43fd81fd934e26465ea16762a0a97d7d/python-frontend/src/main/java/org/sonar/python/semantic/SymbolUtils.java#L134